docs/docs/migration-guide.mdx

```diff
@@ -126,6 +126,152 @@ confidences. This change can affect the maximum confidence predicted by the
 However, the previous behavior can still be enforced by setting
 `ranking_length=10` and `renormalize_confidences=True`.
 
+### Slot Mappings
+
+As of version 3.0, there is a single explicit mechanism for slot filling enabled by defining slot mappings for each slot
+in the `slots` section of the domain file.
+
+This mechanism replaces the implicit slot setting via auto-fill of slots with entities of the same name.
+The `auto_fill` key in the domain is no longer available, as well as the `auto_fill` parameter in the constructor of
+the `Slot` class.
+
+While forms continue to request the next slot, the extraction of slots is now delegated exclusively to the new default
+action `action_extract_slots`. This action runs in the background after each user turn without it being required to be
+triggered manually.
+
+This approach allows slots to be kept up to date over the course of a conversation, and removes duplicated effort in
+mapping the same slots in multiple forms.
+
+You are now required to add the `mappings` key for each slot in the 3.0 domain format, including any slots with custom
+mappings. The keys used to define predefined mappings in 2.0 were maintained. Additionally, you can define slots with
+custom mappings implemented in a custom action, like so:
+
+```yaml
+slots:
+  is_existing_customer:
+    type: bool
+    mappings:
+    - type: custom
+      action: action_verify_customer_status
+```
+
+You can also create a custom validation action extending the `ValidationAction` in the Rasa SDK to either
+define validation methods for slots with predefined mappings or custom extraction and validation methods for slots
+with custom mappings. In either cases, you do not need to specify the `action` key in the mapping, however you will
+need to list `action_validate_slot_mappings` in the domain `actions`.
+You can continue using `FormValidationAction` as in 2.0, however please note this will only run if the form is activated.
+
+Because slot mappings are now relocated from `forms` to `slots`, the `required_slots` of a form has been converted in a
+list of slot names.
+
```

Comment:
I would add an extra header here to called something like #### Automatic slot migration from 2.0 to 3.0 to make it easier to skim